pallets/handles/src/suffix.rs

```diff
@@ -0,0 +1,97 @@
+//! # Suffix Generator
+//!
+//! `suffix_generator` provides a `SuffixGenerator` struct to generate unique suffix sequences for a given range
+//! and seed, excluding already used suffixes.
+
+use core::hash::Hasher;
+use rand::{rngs::SmallRng, Rng, SeedableRng};
+use sp_std::vec::Vec;
+use twox_hash::XxHash64;
+
+/// A generator for unique suffix sequences.
+///
+/// Given a min, max range, and a seed, generates unique suffix sequences by excluding
+/// already used suffixes.
+pub struct SuffixGenerator {
+	min: usize,
```

Comment:
Do we want to use `usize` for all of this?

Reply:
The only idea behind keeping usize was to make it compatible with any arch (as previously these were being moved to general purpose crate)

We can strip it down to `u16` , what you suggest?